deepchem/models/tensorgraph/layers.py

```diff
@@ -480,6 +484,24 @@ def _get_scope_name(self):
       return self._shared_with._get_scope_name()
 
 
+class BPGather2(Layer):
```

Comment:
why 2?  What is different?  Do we still need BPGather?

Reply:
Needs a docstring.

Reply:
Also, let's rename to remove the 2 :)

Reply:
For compatibility purposes, the older BPGather is referenced by legacy parts of the code. I'd be happy to add a docstring though.